message-format.md

```diff
@@ -1,47 +1,106 @@
-
-\*Required for Prime Fimp messages.
+# Message Format
+
+Messages sent over MQTT adhering to the FIMP protocol are JSON messages which meet schema requirements outlined in this document.
+
+## Example message
+
+Following is the example of a FIMP message containing energy consumption report from an electricity meter. More examples can be found in particular [services](services/services.md)
+specifications.
+
+```json
+{
+  "serv": "meter_elec",
+  "type": "evt.meter.report",
+  "val_t": "float",
+  "val": 255.488998413086,
+  "storage": {
+    "sub_value": "kWh"
+  },
+  "props": {
+    "delta_t": "120",
+    "prv_data": "255.488998",
+    "direction": "import",
+    "unit": "kWh"
+  },
+  "tags": [],
+  "uid": "e604e951-7afb-4f96-981b-62e905757686",
+  "ctime": "2022-12-02T10:08:27.5+01:00",
+  "src": "zwave-ad",
+  "ver": "1",
+  "topic": "pt:j1/mt:evt/rt:dev/rn:zw/ad:1/sv:meter_elec/ad:7_0"
+}
+```
+
+## Message Properties
+
+Following is the list of message properties in accordance to FIMP version 1.
+
+| Property  | Type         | Example                                   | Required | Description                                                                                                                                                                                                                          |
+|-----------|--------------|-------------------------------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `serv`    | String       | `"out_bin_switch"`                        | Yes      | The name of the service the interface is part of. Service names should utilize `snake_case`.                                                                                                                                         |
+| `type`    | String       | `"evt.binary.report"`                     | Yes      | The name of the interface represented by the message. See the [Interface Format](#interface-format) section for more details.                                                                                                        |
+| `val_t`   | String       | `"bool"`                                  | Yes      | Type of the value stored in `val` property. See the list of [Value Types](#value-types) section for more details.                                                                                                                    |
+| `val`     | Any          | `3.33`                                    | Yes      | Payload of the message of which type is defined in `val_t` property.                                                                                                                                                                 |
```

Comment:
1) Payload of the message with type is.. (remove redundant 'of')
2) Any is not a type just placeholder. Maybe use __